spatial/transform/umeyama.go

```diff
@@ -0,0 +1,144 @@
+// Copyright ©2025 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package transform
+
+import (
+	"errors"
+
+	"gonum.org/v1/gonum/mat"
+	"gonum.org/v1/gonum/stat"
+)
+
+const (
+	// ErrMsgSVDFailed is the error message for SVD factorization failure.
+	ErrMsgSVDFailed = "transform: SVD factorization failed"
+)
+
+// Umeyama estimates the similarity transformation parameters between two matrices X and Y.
+//
+// This is an implementation of the algorithm presented in:
+// "Least-Squares Estimation of Transformation Parameters Between Two Point Patterns"
+// by Shinji Umeyama, IEEE Transactions on Pattern Analysis and Machine Intelligence,
+// Vol. 13, No. 4, April 1991, which can be found here: https://doi.org/10.1109/34.88573
+//
+// The algorithm finds the optimal similarity transformation [c, R, t] ∈ Sim(m)
+// that minimizes the mean squared error between two point patterns.
+//
+// The transformation relates the point sets as:
+// Y ≈ c * R * X + t
+//
+// The dimensions of X and Y must be equal. The function will panic if they are not.
+// The points require consistent indexing. This means that point i of X needs to correspond
+// to point i of Y.
+//
+// In this implementation, rows represent points and columns represent dimensions.
+//
+// Umeyama returns the scale factor c, the rotation matrix r and the translation vector t.
+// If a computation fails, Umeyama will return an error.
+// varThreshold is used for detecting a degenerate input by comparing it with the variance of x. This is necessary
+// because a variance equal or close to zero could cause numerical instability and/or division by zero.
+// In case of variance <= threshold, Umeyama will return an error.
+// The threshold should be >= 0. If a negative value is passed, the default threshold of 1e-10 will be used.
+func Umeyama(x, y *mat.Dense, varThreshold float64) (c float64, r *mat.Dense, t *mat.VecDense, err error) {
```

Comment:
The only error this can return is for when the SVD fails, so this can just be `Umeyama(x, y *mat.Dense, varThreshold float64) (c float64, r *mat.Dense, t *mat.VecDense, ok bool)` with appropriate changes in the body.